routes/index.js

```diff
@@ -236,3 +193,84 @@ var checkAuth = function (req, res, next) {
+  *********************************************************************************************************/
+function getStnMngmnt(req, res) {
+  //set filter based on query parameter passed in
+  var filter = req.params.filter;
+  filter = (filter === "registered") ? true : (filter === "unregistered") ? false : undefined;
+
+  var data = {
+    messageType: req.messageType,
+    message: req.message
+  }
+
+  //get results from database that match filter
+  dbAPI.getStations(filter).then(function(ret) {
+    if (ret !== false && ret !== undefined) {
+      ret.sort(function(a,b) {
+        return (a.registered === b.registered) ? 0 : a.registered ? 1 : -1;
+      });
+      //display sorted array of stations, unregistered stations first
+      data.obj = ret;
+    } else {
+      //display error to user if no database using alert message on page
+      data.obj = ret;
+      data.messageType = 'error',
+      data.message = "There was a problem communicating with the database.\
+                      Please contact the DB administrator."
+    }
+    res.render('stationManagement.njk', data);
+  });
+}
+
+
+/*postStnMngr serves two purposes: if req.body JSON contains a "delete" element, then the
+ *     user pushed a "delete" button and the station referenced by "sId" element is deleted from tables.
+ *     Otherwise, the station identified by "sId" is updated to values in JSON body and registered = true.
+ *     Implements message passing: signals success or failure to "next" in "req.messageType" and "req.message"
+ ***************************************************************************************************************/
+function postStnMngr(req, res, next) {
+  if (!req.body) {
+    //400 Bad Request
+    return res.sendStatus(400);
+  }
+
+  req.messageType = "success"; //temporarily assume success!
+
+  if(req.body.delete === "true") {
+    //delete station referenced by sId in body
+    dbAPI.deleteStation(req.body.sId).then(function(ret) {
+      if (ret.result === true) {
+        req.message = req.body.name + " has been successfully deleted.";
+      } else {
+        req.messageType = "error";
+        req.message = "Internal problem - unable to delete " + req.body.name + ".";        
+      }
+      next();
+    });
+  } else {
+    //convert registered string to a boolean & give new update date
+    req.body.registered = (req.body.registered === "true") ? true : false;
+    req.body.updatedAt = new Date();
+
+    //update station referenced by sId in body
+    dbAPI.modifyStation(req.body.sId, req.body).then(function (ret) {
+      //prepare to pass success message to next()
+      if (ret.result === true) {
+        //req.message = "Successfully updated " + req.body.name + ".";
+        req.message = req.body.name + " has been succesfully updated.";
+      } else {
+        req.messageType = "error";
+        req.message = "Internal problem - unable to update " + req.body.name + ".";
+      }
+      next();
```

Comment:
See note above - we are passing the database's success or failure on so the "next" router can handle it.